### PR TITLE
refactor: Anki pages WebViewClient creation and asset loading

### PR DESCRIPTION
**Pull request by BrayanDSO**
## Purpose / Description
In the commits.

## Approach
In the commits. 
## How Has This Been Tested?

Emulator 31: Open the statistics and card info pages to see if things are still loading

## Checklist
_Please, go through these checks before submitting the PR._

- [X] You have a descriptive commit message with a short title (first line, max 50 chars).
- [X] You have commented your code, particularly in hard-to-understand areas
- [X] You have performed a self-review of your own code
- [ ] UI changes: include screenshots of all affected screens (in particular showing any new or changed strings)
- [ ] UI Changes: You have tested your change using the [Google Accessibility Scanner](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor)

### CODE REVIEW

**Review by Mukuljangir372 — APPROVED** [review]

File: AnkiDroid/src/main/java/com/ichi2/anki/pages/PageWebViewClient.kt [hunk 1]
```diff
@@ -35,6 +38,23 @@ open class PageWebViewClient : WebViewClient() {
...
+        request: WebResourceRequest
+    ): WebResourceResponse? {
+        val path = request.url.path
+        if (request.method == "GET" && path?.startsWith("/backend/web") == true) {
+            try {
+                val mime = AssetHelper.guessMimeType(path)
+                val inputStream = view.context.assets.open(path.substring(1))
+                return WebResourceResponse(mime, null, inputStream)
+            } catch (_: IOException) {
+                Timber.w("%s not found", path)
```
Comment (david-allison) [inline AnkiDroid/src/main/java/com/ichi2/anki/pages/PageWebViewClient.kt:52]: This leaks user file names to ACRA

**Review by david-allison — CHANGES_REQUESTED** [review]
Much cleaner! One request to not log the file name

File: AnkiDroid/src/main/java/com/ichi2/anki/pages/PageWebViewClient.kt (hunk 1 above)
Comment (BrayanDSO) [reply AnkiDroid/src/main/java/com/ichi2/anki/pages/PageWebViewClient.kt:52]: this only applies to files under `/backend/web/` that are used in Anki backend pages, so there shouldn't be any user files there

Comment (david-allison) [reply AnkiDroid/src/main/java/com/ichi2/anki/pages/PageWebViewClient.kt:52]: We're almost certainly fine here then

**Review by david-allison — APPROVED** [review]

File: AnkiDroid/src/main/java/com/ichi2/anki/pages/PageWebViewClient.kt (hunk 1 above)
Comment (dae) [reply AnkiDroid/src/main/java/com/ichi2/anki/pages/PageWebViewClient.kt:52]: (logging the backend names is helpful for debugging issues with missing files, like the recent missing anki-package-import.html problem)

**Review by lukstbit — APPROVED** [review]
LGTM!

Left two small issues(one from this refactoring + one from my code) if you want to fix them here. Either way feel free to merge this.

File: AnkiDroid/src/main/java/com/ichi2/anki/pages/AnkiPackageImporterFragment.kt [hunk 2]
```diff
@@ -31,11 +31,9 @@ class AnkiPackageImporterFragment : PageFragment() {
         get() = resources.getString(R.string.menu_import)
     override val pageName: String
         get() = "import-anki-package"
-    override lateinit var webViewClient: PageWebViewClient
-    override var webChromeClient: PageChromeClient = PageChromeClient()
     private lateinit var backCallback: OnBackPressedCallback
```
[the remark this answers is not in the document]
Comment (BrayanDSO) [reply AnkiDroid/src/main/java/com/ichi2/anki/pages/AnkiPackageImporterFragment.kt]: done

File: AnkiDroid/src/main/java/com/ichi2/anki/pages/PageFragment.kt [hunk 3]
```diff
@@ -44,20 +44,40 @@ abstract class PageFragment : Fragment(), PostRequestHandler {
         container: ViewGroup?,
         savedInstanceState: Bundle?
     ): View? {
-        val view = inflater.inflate(R.layout.page_fragment, container, false)
+        return inflater.inflate(R.layout.page_fragment, container, false)
```
[the remark this answers is not in the document]
Comment (BrayanDSO) [reply AnkiDroid/src/main/java/com/ichi2/anki/pages/PageFragment.kt]: TIL. Really nice. Thanks!

**Pull request merged**